documentation/doxygen/converttonotebook.py

```diff
@@ -416,90 +474,70 @@ def split(text):
     ... }''')
     ('void tutorial(){\\n   content of tutorial\\n}', ['\\n# <markdowncell>\\n  This is a multiline\\n description of the\\n helper function\\n \\n# <codecell>\\n%%cpp -d\\nvoid helper(arguments = values){\\n   helper function\\n   content spans lines\\n}'], '')
     """
-    functionReString="("
-    for cpptype in gTypesList:
-        functionReString += ("^%s|") % cpptype
-
-    functionReString = functionReString[:-1] + r")\s?\*?&?\s?[\w:]*?\s?\([^\)]*\)\s*\{.*?^\}"
-
-    functionRe = re.compile(functionReString, flags = re.DOTALL | re.MULTILINE)
-    #functionre = re.compile(r'(^void|^int|^Int_t|^TF1|^string|^bool|^double|^float|^char|^TCanvas|^TTree|^TString|^TSeqCollection|^Double_t|^TFile|^Long64_t|^Bool_t)\s?\*?\s?[\w:]*?\s?\([^\)]*\)\s*\{.*?^\}', flags = re.DOTALL | re.MULTILINE)
-    functionMatches = functionRe.finditer(text)
+    functionRe = re.compile(r"[^(/;]*?\s+[*&]*([\w:]+)\s*\(", flags = re.DOTALL | re.MULTILINE)
+    tailRe = re.compile(r"}\s*;?", flags = re.DOTALL | re.MULTILINE)
     helpers = []
+    definitions = []
     main = ""
-    for matchString in [match.group() for match in functionMatches]:
-        if tutName == findFunctionName(matchString):  # if the name of the function is that of the macro
-            main = matchString
+    searchStart = 0
+    for curlyBegin,curlyEnd in findBracedBlock(text, 0, "{"):
+        functionMatches = [match for match in functionRe.finditer(text, searchStart, curlyBegin)] 
+        
+        tailMatch = tailRe.match(text, curlyEnd)
+        if tailMatch:
+            curlyEnd = tailMatch.end()
```

Comment:
Do we need to match `tailRe` here?  Shouldn't `findBracedBlock()` guarantee that `curlyEnd` always matches the closing brace at `curlyBegin`?

Reply:
I don't remember.